src/jobsautoreport/slack.py

```diff
@@ -14,87 +14,126 @@ def __init__(self, web_client: WebClient, channel_id: str) -> None:
         self._client = web_client
         self._channel_id = channel_id
 
-    def send_report(self, report: Report) -> None:
-        blocks = self._format_message(report)
-        logger.debug("Message format finished successfully")
+    def _post_message(
+        self,
+        report: Report,
+        format_function: Callable[[Report], list[dict[str, Any]]],
+        thread_time_stamp: Optional[str],
+    ) -> str:
+        blocks = format_function(report)
+        logger.debug("Message formatted successfully")
         response = self._client.chat_postMessage(
-            channel=self._channel_id, blocks=blocks
+            channel=self._channel_id, blocks=blocks, thread_ts=thread_time_stamp
         )
         response.validate()
         logger.info("Message sent successfully")
+        return response["ts"]
 
-        time_stamp = response["ts"]
-        blocks = self._format_presubmit_comment(report)
-        logger.debug("comment format finished successfully")
-        response = self._client.chat_postMessage(
-            channel=self._channel_id, blocks=blocks, thread_ts=time_stamp
+    def _create_and_upload_file(
+        self,
+        jobs: list[tuple[str, Any]],
+        file_title: str,
+        thread_time_stamp: Optional[str],
+        top_triggered: bool,
+    ) -> None:
+        filename = file_title.replace(" ", "_").lower()
+        file_path = f"/tmp/{filename}.png"
+        initial_comment = file_title
+        if top_triggered:
+            self._create_quantity_image(file_path, jobs)
+        else:
+            self._create_success_image(file_path, jobs)
+        response = self._client.files_upload(
+            channels=[self._channel_id],
+            file=file_path,
+            filename=filename,
+            initial_comment=initial_comment,
+            thread_ts=thread_time_stamp,
         )
         response.validate()
-        logger.info("Presubmit comment sent successfully")
-        blocks = self._format_equinix_comment(report)
-        logger.debug("comment format finished successfully")
-        response = self._client.chat_postMessage(
-            channel=self._channel_id, blocks=blocks, thread_ts=time_stamp
+        logger.info("%s was uploaded successfully", filename)
+
+    def send_report(self, report: Report) -> None:
+        thread_time_stamp = self._post_message(
+            report,
+            self._format_message,
+            thread_time_stamp=None,
+        )
+        self._create_and_upload_file(
+            report.top_10_failing_e2e_or_subsystem_periodic_jobs,
+            file_title="Top 10 Failed Periodic Jobs",
+            thread_time_stamp=None,
```

Comment:
why it's not in the thread?

Reply:
Initially it was supposed to be one message for `periodic` jobs, and all others (`presubmit` jobs and `equinix`0 jobs were supposed to be in comments, but when we switched to files I can attach a file to a message, I have to upload it as a different message. It might be a good idea to put everything as a comment. WDYT?

Reply:
Either way it's not critical. For now do whatever is easier and we can optimize later